crates/biome_js_analyze/src/lint/nursery/no_yoda_expression.rs

```diff
@@ -0,0 +1,412 @@
+use crate::{utils::is_node_equal, JsRuleAction};
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make::{self, js_binary_expression, token};
+use biome_js_syntax::{
+    AnyJsExpression, AnyJsLiteralExpression, AnyJsStatement, JsBinaryExpression, JsBinaryOperator,
+    JsLanguage, JsLogicalExpression, JsLogicalOperator, JsSyntaxKind, JsUnaryOperator,
+    JsYieldArgument, JsYieldExpression, T,
+};
+use biome_rowan::{AstNode, BatchMutationExt, NodeOrToken, SyntaxTriviaPiece, TriviaPieceKind};
+
+declare_rule! {
+    /// Disallow the use of yoda expressions.
+    ///
+    /// A Yoda expression is a programming style where, given a binary operation, the "static" part of the binary operation is placed on the left-hand side.
+    /// This rule **forbids** the use of Yoda expressions and enforces the placing of the "static" part of the binary operations on the right-hand side.
+    ///
+    /// ## Exceptions
+    ///
+    /// Range expressions like `0 < value && value < 1` or `value <= 0 || 1 < value` are allowed.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// if ("red" == value) {}
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (true === value) {}
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (5 != value) {}
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// if (value === "red") {}
+    /// ```
+    ///
+    /// ```js
+    /// if (value === value) {}
+    /// ```
+    ///
+    /// ```js
+    /// if (value != 5) {}
+    /// ```
+    ///
+    /// ```js
+    /// if (0 < value && value < 1) {}
+    /// ```
+    ///
+    /// ## Resources
+    /// - [Wikipedia definition](https://en.wikipedia.org/wiki/Yoda_conditions)
+    ///
+    pub NoYodaExpression {
+        version: "next",
+        name: "noYodaExpression",
+        language: "js",
+        sources: &[RuleSource::Eslint("yoda")],
+        recommended: false,
+        fix_kind: FixKind::Safe,
+    }
+}
+
+impl Rule for NoYodaExpression {
+    type Query = Ast<JsBinaryExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let left = node.left().ok()?;
+        let right = node.right().ok()?;
+
+        let has_yoda_expression = node.is_comparison_operator()
+            && is_literal_expression(&left).unwrap_or_default()
+            && !is_literal_expression(&right).unwrap_or_default()
+            && !is_range_assertion(node).unwrap_or_default();
```

Comment:
```suggestion
            && is_literal_expression(&left)?
            && !is_literal_expression(&right)?
            && !is_range_assertion(node)?;
```

Reply:
I was able to remove all but this call: `!is_range_assertion(node).unwrap_or_default()`, this helper might return `None` but the code should keep running in that case, is there another approach I could take here? I can convert it to return `bool` instead of `Option<bool>` if you think it's better

Reply:
Yeah, in this case, it makes sense to return `bool`. Some documentation could help explain why

Reply:
The method is now returning `bool`, and also, I did my best to explain in details what this method does, let me know what you think :)